src/pages/home-page.js

```diff
@@ -46,41 +53,107 @@ export default function HomePage() {
     setPokemonsDisplay(prePokemons => [...prePokemons, ...pokemonSlice])
   }, [pokemons, offset])
 
+  // Start search when form submit.
+  const handleSubmitSearchForm = e => {
+    e.preventDefault()
+    if (keywords.trim().length > 0) {
+      const stn = 'name'
+      const stt = 'type'
+
+      // check is search by NAME or TYPE
+      if (searchType.type === stn) {
+        const resultFilter = pokemons.filter(pokemon => {
+          return pokemon.name.trim().includes(`${keywords}`.trim())
+        })
+        setPokemonsDisplay(resultFilter)
+      }
+      if (searchType.type === stt) {
+        // get all types.
+        axios
+          .get('https://pokeapi.co/api/v2/type/')
+          .then(res => res.data.results)
+          .then(types => {
+            return types.filter(type => {
+              return type.name.trim().includes(`${keywords}`.trim())
+            })
+          })
+          // filter type and get all pokemon with type name.
+          .then(resultTypeFilter => {
+            axios
+              .get(`${resultTypeFilter[0]?.url}`)
+              .then(response => response.data)
+              .then(typeDetail => {
+                const result = typeDetail?.pokemon.map(
+                  pokemon => pokemon.pokemon
+                )
+                // set result filter for display pokemon.
+                setPokemonsDisplay(result)
+              })
+          })
+      }
+    }
+  }
   return (
     <div className="container">
       {/* Logo */}
       <h1 className="text-center mb-5" id="logo-page">
         Pokemon Dex
       </h1>
       {/* ! Search Form*/}
-      <form id="form-search" className="mb-5">
-        <label htmlFor="search-pokemon-input" className="form-label">
-          Search pokemon
-        </label>
-        <input
-          type="text"
-          className="form-control"
-          id="search-pokemon-input"
-          aria-describedby="search-pokemon-input"
-        />
+      <form id="form-search" className="mb-5" onSubmit={handleSubmitSearchForm}>
+        <div className="row">
+          <div className="col-10">
+            <label htmlFor="search-pokemon" className="form-label">
+              Search
+            </label>
+            <input
+              type="text"
+              className="form-control"
+              id="search-pokemon"
+              aria-describedby="search-pokemon"
+              onChange={e => {
+                setKeywords(e.target.value)
+
+                // reset pokemons while input search form is empty.
+                if (e.target.value.trim() === '') {
+                  const pokemonSlice = pokemons.slice(0, 0 + 20)
+                  setPokemonsDisplay(pokemonSlice)
+                }
+              }}
```

Comment:
please define a handler function for this one instead of inline syntax

Reply:
i will fix that